iib/workers/tasks/build_regenerate_bundle.py

```diff
@@ -458,3 +492,28 @@ def _adjust_csv_annotations(operator_csvs, package_name, org_csv_annotations):
             csv_annotations[annotation] = value
 
         operator_csv.dump()
+
+
+def _apply_repo_enclosure(bundle_metadata, org_enclose_repo_namespace, org_enclose_repo_glue):
+    """
+    Apply repo_enclosure customization to the bundle image.
+
+    :param dict bundle_metadata: the dictionary of CSV's and relatedImages pull specifications
+    :param str org_enclose_repo_namespace: the string sprecifying the namespace of the modified
+        pull specs in the CSV files
+    :param str org_enclose_repo_glue: the string specifying the enclosure glue to be applied
+        to modify the pull specs in the CSV files
+    """
+    replacement_pullspecs = {}
+    for pullspec in bundle_metadata['found_pullspecs']:
+        new_pullspec = ImageName.parse(pullspec.to_str())
+
+        repo_parts = new_pullspec.repo.split('/')
+        if org_enclose_repo_namespace != new_pullspec.namespace:
+            repo_parts.insert(0, new_pullspec.namespace)
+
+        new_pullspec.namespace = org_enclose_repo_namespace
+        new_pullspec.repo = org_enclose_repo_glue.join(repo_parts)
+        replacement_pullspecs[pullspec] = new_pullspec
```

Comment:
It would be nice to log this. How about if we just change `_replace_csv_pullspecs` to iterate through `replacement_pullspecs` at the very beginning to log what will be replaced with what. Then we don't have to ensure ever caller is logging this information as well, i.e. `_apply_registry_replacements` and `_resolve_image_pull_specs`

Reply:
Done.